### PR TITLE
Scripts: Update `puppeteer-core` dependency

### PR DESCRIPTION
**Pull request by jacobcassidy**
## What?
Upgrades the `puppeteer-core` package to the latest version (23.1.0).

## Why?
This PR fixes the issue with `@wordpress/scripts` having five high-severity vulnerabilities introduced with an older version of the `puppeteer-core` package.

See: #63771

## How?
Removes the sub-dependencies versions with vulnerabilities.

## Testing Instructions
1. In a WP theme development environment, run `npm install @wordpress/scripts path webpack-remove-empty-scripts --save-dev`
2.  Run `npm audit` and you'll see a warning for 5 high-severity vulnerabilities.
3. Add the following to your package.json file: 
    ```json 
    "overrides": {
        "puppeteer-core": "^23.1.0"
    }
    ```
5. Run `npm install` to update the packages.
6. The vulnerabilities are now removed.

### CODE REVIEW

File: packages/scripts/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@wordpress/scripts",
-	"version": "28.5.0",
+	"version": "28.6.0",
```
Comment (gziolo) [inline packages/scripts/package.json]: `version` change is automated so this line should get reverted:

```suggestion
	"version": "28.5.0",
```

File: packages/scripts/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
 
 ## Unreleased
 
+## 28.6.0 (2024-08-17)
```
Comment (gziolo) [inline packages/scripts/CHANGELOG.md]: Changelog changes are automated so this should get updated to:

```suggestion
### Bug Fixes
```

**Review by gziolo — COMMENTED** [review]
Thank you for opening this issue. There are some details to polish related to the packages release process. I also see some CI issues reported that need to be further investigated. It looks like the changes to the `package-lock.json` will require some tweaks to make it work with the monorepo.

@Mamaduka and @swissspidy – do we still use Puppeteer for e2e tests in any place in Gutenebrg or WordPress core? What's the plan with the scripts powered by Puppeteer? How can we test these changes?

**Comment by Mamaduka** [comment]
@gziolo, we do not, but it needs to be removed in a backward-compatible manner. See #60357.

**Review by gziolo — APPROVED** [review]
Thank you for the follow-up commits. It's good to go.

**Comment by jacobcassidy** [comment]
@gziolo Thanks for the instructions and follow-up.

**Comment by gziolo** [comment]
It looks like ESLint detected that `puppeteer-core` doesn't get installed in the top `node_modules` folder. To fix it, we would have to put `puppeteer-core` as a dev dependency in the main `package.json` file. Remove all changes added in this branch to the main `package-lock.json` file. Run `npm install`, and it should all be good.

**Pull request merged**